File: 119-pascals-triangle-ii/pascals-triangle-ii.py
class Solution(object):
    def getRow(self, rowIndex):
        main = [[1],[1,1]]
        if(rowIndex == 0):
            return [1]
        
        elif(rowIndex == 1):
            return [1,1]

        for i in range(1, rowIndex+1):
            temp = [1]
            for j in range(1,len(main[-1])):
                temp.append(main[-1][j-1]+main[-1][j])
            temp.append(1)
            main.append(temp)
        return main[-2]

        

File: 119-pascals-triangle-ii/pascals-triangle-ii.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        if (rowIndex == 0) {
            return new ArrayList<>(Arrays.asList(1));
        } else if (rowIndex == 1) {
            return new ArrayList<>(Arrays.asList(1, 1));
        }

        List<List<Integer>> l = new ArrayList<>();

        for (int i = 0; i <= rowIndex; i++) {
            l.add(new ArrayList<>());

            for (int j = 0; j <= i; j++) {
                l.get(i).add(1);
            }
        }

        for (int i = 2; i <= rowIndex; i++) {
            for (int j = 1; j < i; j++) {
                l.get(i).set(j,
                    l.get(i - 1).get(j - 1) + l.get(i - 1).get(j));
            }
        }

        return l.get(rowIndex);
    }
}